qualang_tools/plot/Fitting/fitting_functions.py

```diff
@@ -0,0 +1,348 @@
+from scipy import optimize
+import numpy as np
+
+
+def fit_ramsey(x, y):
+    """
+    Create a fit to Ramsey experiment of the form
+
+    .. math::
+    f(x) = uncertainty_population * (1 - np.exp(-x * (1/tau))) + amp / 2 * (
+        np.exp(-x * (1/tau))
+        * (initial_offset * 2 + np.cos(2 * np.pi * f * x + phase))
+        )
+
+    for unknown parameters `f`, `phase`, `tau`, `amp`, `uncertainty_population` and `initial_offset`.
+
+    :param x:  The data on the x axis
+    :param y: The data on the y axis
+    :return: A dictionary of (fit_func, f, phase, tau, amp, uncertainty_population, initial_offset)
+        fit_func - The fitted function
+        f -  The detuning frequency
+        phase - The phase
+        tau - The decay constant
+        amp - The amplitude
+        uncertainty_population -  The uncertainty in population
+        initial_offset - The initial offset
```

Comment:
I can change tau to be T2, is it more clear? the uncertainty population is actually the amplitude.. and the initial offset is not zero because of noise I guess... ?

Reply:
I changed the comments :)

Reply:
If tau is actually T2, then yes I would call it T2
In general I think it is best to use physics related quantities as fitting parameters so that users know directly what they represent.
It is still not clear to me what uncertainty_population represent... Where does this equation come from?

Reply:
I will talk to Asaf and update it there :)